### PR TITLE
 Creation of the script classification,adapt test file and solve dependencies problem

### PR DESCRIPTION
**Pull request by Melviyn**
Creation of the script allows to:
- Execute the classification following the model mapuce on buildings of a particular city
- Create the table that contains the aggregation of each typology of buildings geometry(THE_GEOM field)
- Creates the table containing the majority and secondary typology by USR

The mapuce model is now download [here](https://github.com/Melviyn/MApUCE_tools/raw/dev/model/mapuce-1.0.model) don't need to get the .model ourselves

Problem with dependencies Weka OSGI resolve, you must download and run OSGI-Weka on OrbisGIS before launching the plugin Mapuce_Tools

The test that are now on less data for a better understanding of the test (iris data)

### CODE REVIEW

File: src/main/resources/org/orbisgis/mapuce/scripts/MApUCE/classification.groovy [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    "u.B_STD_H_MEAN AS u_bH_STD_MEAN,u.B_M_NW_COMPACITY AS u_bCOMP_NWMEAN,u.B_M_W_COMPACITY AS u_bCOMP_WMEAN,u.B_STD_COMPACITY AS u_bCOMP_STD,u.DIST_TO_CENTER AS u_DIST_CENTER,"+
+    "u.BUILD_DENS AS u_BUILD_DENS,u.VEGET_DENS AS u_VEG_DENS,u.ROAD_DENS AS u_ROAD_DENS,u.EXT_ENV_AREA AS u_FWALL_A "+
+    "FROM BUILDING_INDICATORS AS A "+
+    "JOIN USR_INDICATORS AS U ON a.PK_USR = u.PK "+
+    "JOIN BLOCK_INDICATORS b "+
+    "ON a.PK_USR = b.PK_USR AND a.PK_BLOCK = b.PK_BLOCK"
+    
+
+  File file = File.createTempFile("mapuce", ".model")
+  FileUtils.copyURLToFile(new URL("https://github.com/Melviyn/MApUCE_tools/raw/dev/model/MApUCE-1.0.model"), file)
```
Comment (ebocher) [inline src/main/resources/org/orbisgis/mapuce/scripts/MApUCE/classification.groovy]: What about if the file already exists ?

Comment (Melviyn) [inline src/main/resources/org/orbisgis/mapuce/scripts/MApUCE/classification.groovy]: createTempFile create a File like mapuce+RandomNumber(19 digit)+.model, this is really unlucky if this combinaison exists but for safety i can remove this file at the end of the script.

Comment (SPalominos) [inline src/main/resources/org/orbisgis/mapuce/scripts/MApUCE/classification.groovy]: Before merging, you should change the file path `https://github.com/Melviyn/MApUCE_tools/raw/dev/model/MApUCE-1.0.model` into `https://github.com/orbisgis/MApUCE_tools/raw/master/model/mapuce-1.0.model`

File: src/main/resources/org/orbisgis/mapuce/scripts/MApUCE/classification.groovy [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  ClassifyData cla = new ClassifyData(file.getAbsolutePath(),sql.createConnection())
+ 
+  Statement sta = sql.createConnection().createStatement()
+  ResultSet rs = sta.executeQuery(req) 
+
+  //Transform ResultSet in Instances  
+  logger.warn "Transform ResultSet in Instances"
```
Comment (SPalominos) [inline src/main/resources/org/orbisgis/mapuce/scripts/MApUCE/classification.groovy]: Maybe `logger.warn()` is too worrying for simple users. As the messages are only here to tell the user about the script progression `logger.info()` may be better.

File: src/main/resources/org/orbisgis/mapuce/scripts/MApUCE/classification.groovy [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  sql.execute "DROP TABLE IF EXISTS USR_TYPO_WITH_NULL_VALUE"
+
+
+  logger.warn "Done"
+}
+/** String output of the process. */
+@LiteralDataOutput(
+        title="Output message",
+        resume="The output message")
+String literalOutput
```
Comment (SPalominos) [inline src/main/resources/org/orbisgis/mapuce/scripts/MApUCE/classification.groovy:100]: The variable `literalOutput` is never assign. You should give it a value (even if it is just something like "Work done.") as each WPS process need to have an output.

File: src/main/resources/org/orbisgis/mapuce/scripts/MApUCE/import_zones.groovy [hunk 4]
```diff
@@ -29,6 +27,7 @@ if(!login.isEmpty()&& !password.isEmpty()){
 
 	query+="'"+password+"', '"+schemaFromRemoteDB+"', "
 	query+= "'"+tableFromRemoteDB+"')";
+        sql.execute "DROP TABLE IF EXISTS COMMUNE_TEMP"
```
Comment (SPalominos) [inline src/main/resources/org/orbisgis/mapuce/scripts/MApUCE/import_zones.groovy]: Why indenting here ?

File: src/main/resources/org/orbisgis/mapuce/scripts/MApUCE/morphological_indicators.groovy [hunk 5]
```diff
@@ -255,36 +259,8 @@ sql.execute "DROP SCHEMA DATA_WORK"
...
-        resume = "The table to edit",
-        isSpatial = true)
-String buildingsTable
-
-
 /** String output of the process. */
 @LiteralDataOutput(
         title="Output message",
         resume="The output message")
 String literalOutput
```
Comment (SPalominos) [inline src/main/resources/org/orbisgis/mapuce/scripts/MApUCE/morphological_indicators.groovy:267]: The variable `literalOutput` is never assign. You should give it a value (even if it is just something like "Work done.") as each WPS process need to have an output.

File: src/test/java/org/orbisgis/mapuce/randomForest/RandomForestTest.java [hunk 6]
```diff
@@ -84,50 +84,50 @@ public void testCreateModel() throws SQLException, IOException, Exception {
...
     @Test
     public void testCreateInstancesFromDB() throws SQLException, IOException, Exception {
         
-        ClassifyData test = new ClassifyData(this.getClass().getResource("iris.model").getPath(),connection);
+        String modelFile = this.getClass().getResource("iris.model").getPath();
+        ClassifyData test = new ClassifyData(modelFile,connection);
         
         //=================== Create the tables ============
-        String sql = "DROP TABLE IF EXISTS IRIS ";
+        String sql = "DROP TABLE IF EXISTS COMMUNE";
```
Comment (SPalominos) [inline src/test/java/org/orbisgis/mapuce/randomForest/RandomForestTest.java]: It seems to be `IRIS` instead of `COMMUNE`

**Pull request merged**